src/components/vive-controls.js

```diff
@@ -43,36 +45,129 @@ module.exports.Component = registerComponent('vive-controls', {
     this.onButtonDown = function (evt) { self.onButtonEvent(evt.detail.id, 'down'); };
     this.onButtonUp = function (evt) { self.onButtonEvent(evt.detail.id, 'up'); };
     this.onModelLoaded = bind(this.onModelLoaded, this);
+    this.controllerPresent = false;
+    this.everGotGamepadEvent = false;
+    this.lastControllerCheck = 0;
   },
 
-  play: function () {
+  startListening: function () {
     var el = this.el;
+    // console.log('startListening');
     el.addEventListener('buttonchanged', this.onButtonChanged);
     el.addEventListener('buttondown', this.onButtonDown);
     el.addEventListener('buttonup', this.onButtonUp);
     el.addEventListener('model-loaded', this.onModelLoaded);
   },
 
-  pause: function () {
+  stopListening: function () {
     var el = this.el;
+    // console.log('stopListening');
     el.removeEventListener('buttonchanged', this.onButtonChanged);
     el.removeEventListener('buttondown', this.onButtonDown);
     el.removeEventListener('buttonup', this.onButtonUp);
     el.removeEventListener('model-loaded', this.onModelLoaded);
   },
 
-  update: function () {
+  checkIfControllerPresent: function () {
```

Comment:
currently, Oculus Touch (as seen through experimental Chromium) exposes specific gamepad id values for left and right controller, and the hand attribute is set correctly, so although you can't look for the same exact controller (because there is no persistent unique ID for a gamepad exposed by that API) you can look for a Touch controller with the correct hand.

for Vive wands via OpenVR, at present it's a bit ugly since the gamepad id values are not unique, and the hand attribute values are not necessarily correct... so the original code's hardcoding based upon order of appearance is still there.

ideally, the logic that works with Touch could be more generically used, but since with Vive that is broken, makes sense to me to refactor later once things get fixed or a new controller type to support arises.

in the meantime, the current version of PR does provide specific button events in case applications need (the `oculus-touch.A` discussion above) which per our prior discussion I though was worth keeping.  

OK?